Mathlib/RingTheory/Presentation.lean

```diff
@@ -0,0 +1,409 @@
+/-
+Copyright (c) 2024 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.RingTheory.Ideal.Cotangent
+
+/-!
+
+# Presentations of algebras
+
+## Main definition
+- `Algebra.Presentation`: A presentation of a `R`-algebra `S` consists of
+  1. `I`: The type of variables.
+  2. `val : I → S`: The assignment of each variable to a value.
+  3. `σ`: A section of `R[I] → S`.
+- `Algebra.Presentation.Hom`: Given a commuting square
+  ```
+  R --→ P = R[I] ---→ S
+  |                   |
+  ↓                   ↓
+  R' -→ P' = R'[I'] → S
+  ```
+  A hom between `P` and `P'` is an assignment `I → P'` such that the arrows commute.
+- `Algebra.Presentation.Cotangent`: The cotangent spance of a presentation `P = R[I] → S`, I.e. the
+  space `I/I²` with `I` being the kernel of the presentation.
+
```

Comment:
Maybe rename the type of variables from `I` to `ι` (or something else), because it clashes with the name of the kernel?

Reply:
Okay I (finally) renamed it to `vars`.